scripts/utils/get-history-file.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# scripts/utils/get-history-file.sh
+#
+# Returns the session-aware history file path
+
+set -euo pipefail
+base_history_dir="${TMUX_COMPILE_HISTORY_DIR:-$HOME/.tmux-compile-history}"
+session_name="$(tmux display-message -p '#S' 2>/dev/null || echo default)"
+
+# Delete base_dir if it is a file
+if [ -f "$base_history_dir" ]; then
+    rm "$base_history_dir"
```

Comment:
This unconditionally removes a file if TMUX_COMPILE_HISTORY_DIR points to one and then creates a directory. 
That will silently delete a file on a users  machine - Is that intentional? 

Can we instead migrate the file, fail with a clear error, or at least warn the user before removing their file? **Deleting user files without warning seems risky.**